code/controllers/subsystems/mapping.dm

```diff
@@ -78,3 +128,58 @@ SUBSYSTEM_DEF(mapping)
 
 /datum/controller/subsystem/mapping/proc/get_templates_by_category(var/temple_cat) // :33
 	return map_templates_by_category[temple_cat]
+
+// Z-Level procs after this point.
+/datum/controller/subsystem/mapping/proc/get_gps_level_name(var/z)
+	if(z)
+		var/obj/abstract/level_data/level = levels_by_z[z]
+		if(level?.name)
+			return level.get_gps_level_name()
+	return "Unknown Sector"
+
+/datum/controller/subsystem/mapping/proc/increment_world_z_size(var/new_level_type, var/defer_setup = FALSE)
+
+	world.maxz++
+	levels_by_z.len = world.maxz
+	connected_z_cache.Cut()
+
+	if(SSzcopy.zlev_maximums.len)
+		SSzcopy.calculate_zstack_limits()
+	if(!new_level_type)
+		PRINT_STACK_TRACE("Missing z-level data type for z["[world.maxz]"]!")
+		return
+
+	var/obj/abstract/level_data/level = new new_level_type(locate(round(world.maxx*0.5), round(world.maxz*0.5), world.maxz), defer_setup)
+	level.initialize_level()
+	return level
+
+/datum/controller/subsystem/mapping/proc/get_connected_levels(z)
+	if(z <= 0  || z > length(levels_by_z))
+		CRASH("Invalid z-level supplied to get_connected_levels: [z || "NULL"]")
```

Comment:
This incorrectly reports "NULL" instead of 0 if something calls `get_connected_levels(0)`, which is kinda annoying.